code/modules/power/supermatter/supermatter.dm

```diff
@@ -324,30 +325,37 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 				H.hallucination += max(50, min(300, DETONATION_HALLUCINATION * sqrt(1 / (get_dist(mob, src) + 1)) ) )
 			var/rads = DETONATION_RADS * sqrt( 1 / (get_dist(L, src) + 1) )
 			L.rad_act(rads)
-
 	var/turf/T = get_turf(src)
 	for(var/mob/M in GLOB.player_list)
 		if(M.z == z)
 			SEND_SOUND(M, 'sound/magic/charge.ogg')
 			to_chat(M, "<span class='boldannounce'>You feel reality distort for a moment...</span>")
 			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "delam", /datum/mood_event/delam)
-	if(combined_gas > MOLE_PENALTY_THRESHOLD)
+	if(!T)//If something fucks up we just end it. This fix is 4 years old and none ever said why it's here. help.
+		return
+	if(combined_gas > MOLE_PENALTY_THRESHOLD && power > CRITICAL_POWER_PENALTY_THRESHOLD)
+		investigate_log("has collapsed into a resonance cascade.", INVESTIGATE_SUPERMATTER)
+		var/datum/round_event_control/portal_storm_mesa/subtle_i_know
+		if(prob(99))//Clown man bad
+			subtle_i_know = new()
+		else
+			subtle_i_know = new /datum/round_event_control/portal_storm_mesa/clown_storm()
+		subtle_i_know.runEvent()
+		sleep(10)
```

Comment:
Why are you sleeping here?

Reply:
the intent is to have a pause, but on reconsideration the portal storm isn't snappy enough for the effect I was going for.